code/controllers/subsystem/non_firing/SSchangelog.dm

```diff
@@ -204,10 +188,13 @@ SUBSYSTEM_DEF(changelog)
 
 	return data
 
-/datum/controller/subsystem/changelog/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.always_state)
-	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+/datum/controller/subsystem/changelog/ui_state(mob/user)
+	return GLOB.always_state
+
+/datum/controller/subsystem/changelog/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
```

Comment:
This SStgui.try_update_ui, if(!ui), ui = new, ui.open() dance shows up a lot. While there's some subtle differences between implementations I feel like this could be generalized.

Reply:
Can you clarify? You want to roll this behaviour into the base declaration and cut down on the repeated stuff?